packages/@aws-cdk/cdk/lib/aspects/tag-aspect.ts

```diff
@@ -0,0 +1,120 @@
+import { ITaggable, Resource } from '../cloudformation/resource';
+import { IConstruct } from '../core/construct';
+import { Aspect } from './aspect';
+
+export interface TagBaseProps extends TagProperties {
+  /**
+   * The value of the tag
+   */
+  value?: string;
+}
+
+export interface TagProperties {
+  /**
+   * This applies specifically to AutoScalingGroup PropagateAtLaunch
+   */
+  applyToLaunchInstances?: boolean;
+
+  /**
+   * An array of Resource Types that will receive this tag
+   */
+  include?: string[];
+
+  /**
+   * An array of Resource Types that will not receive this tag
+   */
+  exclude?: string[];
+}
+
+/**
+ * The common functionality for Tag and Remove Tag Aspects
+ */
+export abstract class TagBase extends Aspect {
+
+  /**
+   * Test if the construct is a CloudFormation Resource
+   */
+  public static isResource(resource: any): resource is Resource {
```

Comment:
Move this to the Resource class (Resource.isResource is the convention)

Reply:
will fix